### PR TITLE
[IJ plugin] Telemetry: networking

### PR DESCRIPTION
**Pull request by BoD**
Also:
- move `telemetryEnabled` and `hasShownTelemetryOptOutDialog` settings to application level so this is a global and not per project.
- persist project level settings to `workspace.xml` which is typically ignored by VCS (thanks for the tip @wiyarmir!)

### CODE REVIEW

File: intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/refactoring/migration/v3tov4/ApolloV3ToV4MigrationProcessor.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ import com.apollographql.ijplugin.refactoring.migration.v3tov4.item.UpdateMultiM
 import com.apollographql.ijplugin.refactoring.migration.v3tov4.item.UpdateWebSocketReconnectWhen
 import com.intellij.openapi.project.Project
 
-private const val apollo4LatestVersion = "4.0.0-alpha.3"
+private const val apollo4LatestVersion = "4.0.0-beta.1"
```
Comment (martinbonnin) [inline intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/refactoring/migration/v3tov4/ApolloV3ToV4MigrationProcessor.kt:27]: Can this be kept up to date from https://github.com/apollographql/apollo-kotlin/blob/main/scripts/release.main.kts?

Comment (BoD) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/refactoring/migration/v3tov4/ApolloV3ToV4MigrationProcessor.kt:27]: Good idea! 👀

File: intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/telemetry/TelemetryService.kt [hunk 2]
```diff
@@ -95,7 +94,7 @@ class TelemetryService(
 
   private val telemetryEventList: TelemetryEventList = TelemetryEventList()
 
-  private var projectLibraries: Set<Dependency> = emptySet()
+  private var projectLibraries: Set<MavenCoordinates> = emptySet()
 
   private val executor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
```
Comment (martinbonnin) [inline intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/telemetry/TelemetryService.kt:99]: Is this a virtual thread if I run my IntelliJ on Java 21? 🤓

Comment (BoD) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/telemetry/TelemetryService.kt:99]: Haha possibly .... but probably not 😁.

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**